module/modmanager/manager.go

```diff
@@ -64,6 +64,7 @@ func NewManager(
 		packagesDir:             options.PackagesDir,
 		ftdc:                    options.FTDC,
 		modPeerConnTracker:      options.ModPeerConnTracker,
+		failedModules:           make(map[string]bool),
```

Comment:
why do we need another map to track this information? can we add the information into the moduleMap instead?